File: leetcode/0053. 最大子数组和/main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    if (nums.length === 1) {
        return nums[0]
    }
    let max = nums[0]
    for(i = 1; i < nums.length; i++) {
        if (nums[i] < nums[i] + nums[i-1]) {
            nums[i] += nums[i-1]
        }
        if (max < nums[i]) {
            max = nums[i]
        }
    }
    return max
};

File: leetcode/0053. 最大子数组和/main.go
package main

import "fmt"

func main() {
	fmt.Println(maxSubArray([]int{-2, 1, -3, 4, -1, 2, 1, -5, 4}))
}

func maxSubArray(nums []int) int {
	if len(nums) == 1 {
		return nums[0]
	}
	max := nums[0]
	for i := 1; i < len(nums); i++ {
		if nums[i] <= nums[i]+nums[i-1] {
			nums[i] += nums[i-1]
		}
		if nums[i] > max {
			max = nums[i]
		}
	}
	return max
}
